xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -2135,12 +2135,12 @@ void CDVDDemuxFFmpeg::ParsePacket(AVPacket* pkt)
   }
 }
 
-bool CDVDDemuxFFmpeg::IsVideoReady()
+bool CDVDDemuxFFmpeg::IsProgramReady()
```

Comment:
The original intent for this method was to start video with a GOP. IMO renaming it deteriorates the design. Why don't you create an additional method that cares for the audio only case? Then aggregate both into something like IsTsStreamReady

Reply:
OK, will do and push update -- thanks!

Reply:
Actually on second thought, I'm not willing to go this route.  Carrying some variable around to track if there is only an audio stream is IMO worse than the quick-fix this was intended to be. I question why this method didn't reset m_checkvideo to false in the first place for performance reasons, and I don't want to make things any worse here.

If you would be OK with me renaming m_checkprogram to something like m_checktransportstream and IsProgramReady() to is IsTransportStreamReady(), I'm good with that.  Otherwise I think it's best to cancel this PR and open an Issue instead describing the problem so it can be fixed to your standards. This isn't the only thing wrong with demuxing audio-only TS streams via ffmpeg and it may simply be better to say "it's broke, please fix it" than the DIY route.

Thank you for your time.

Reply:
I think I see your point now and it makes sense to do it that way. Is the intention to be more like "if there is a video stream it must be ready AND if there is an audio stream it must be ready"? Then having the two separate functions won't need to care what the other returned.  Like if(IsVideoReady() && IsAudioReady())? If there is no corresponding stream type the individual function returns true instead of false. This may help the problem with audio format changes dropping out sound for a few seconds too. I shall try it tonight.

Reply:
> "if there is a video stream it must be ready AND if there is an audio stream it must be ready"

No, if there is a video stream, audio is not of interest. Remember, the intent is to start with a GOP.

Having IsVideo/AudioReady return true in case there is no video/audio stream is kind of ugly. Those functions should return something like {READY, NOTREADY, NONE}. Then an umbrella function can look like this:

```
bool IsTsStreamReady()
{
  video = IsVideoReady();
  if (video == NOTREADY)
    return false;
  else if (video == NONE)
  {
    audio = IsAudioReady();
    if (audio == NOTREADY)
      false;
  }
  return true;
}
```

> This may help the problem with audio format changes dropping out sound for a few seconds too.

I don't observe this. Most tv recordings switch audio config on every commercial break. I never observe audio drops.

Reply:
How would you feel about the quick-fix version (without the variable/function rename) of this as a PR for Leia and I'll take my time and come up with something more comprehensive for master/Matrix? I'd hate to miss the next dot release for Leia (if there is one) given my limited availability and tendency to come up with things that require so much rework to be acceptable?  I'm currently trying to track down the 'next' bug with audio-only MPEG-TS PVR streams that has to do with visualizations being enabled and won't get back to this PR for a couple/few days